examples/MultiCanvas.cpp

```diff
@@ -303,28 +325,77 @@ void runWg()
+
+    // get adapter and surface properties
+    WGPUFeatureName featureNames[32]{};
+    size_t featuresCount = wgpuAdapterEnumerateFeatures(adapter, featureNames);
+
+    // request device
+    WGPUDevice device{};
+    const WGPUDeviceDescriptor deviceDesc { .label = "The shared device", .requiredFeatureCount = featuresCount, .requiredFeatures = featureNames };
+    auto onDeviceRequestEnded = [](WGPURequestDeviceStatus status, WGPUDevice device, char const * message, void * pUserData) { *((WGPUDevice*)pUserData) = device; };
+    wgpuAdapterRequestDevice(adapter, &deviceDesc, onDeviceRequestEnded, &device);
+
+    // setup surface configuration
+    WGPUSurfaceConfiguration surfaceConfiguration {
+        .device = device,
+        .format = WGPUTextureFormat_BGRA8Unorm,
+        .usage = WGPUTextureUsage_RenderAttachment | WGPUTextureUsage_CopyDst,
+        .width = WIDTH, .height = HEIGHT
+    };
+    wgpuSurfaceConfigure(surface, &surfaceConfiguration);
+
+    // create offscren target texure
+    const WGPUTextureDescriptor textureDesc {
+        .usage = WGPUTextureUsage_CopySrc | WGPUTextureUsage_RenderAttachment,
+        .dimension = WGPUTextureDimension_2D, .size = { SIZE, SIZE, 1 },
+        .format = WGPUTextureFormat_BGRA8Unorm, .mipLevelCount = 1, .sampleCount = 1
+    };
+    WGPUTexture renderTarget = wgpuDeviceCreateTexture(device, &textureDesc);
+
+    WGPUSurfaceTexture surfaceTexture{};
+    wgpuSurfaceGetCurrentTexture(surface, &surfaceTexture);
+
     for (int counter = 0; counter < NUM_PER_LINE * NUM_PER_LINE; ++counter) {
-        auto canvas = tvg::WgCanvas::gen();
+        auto canvas = unique_ptr<tvg::WgCanvas>(tvg::WgCanvas::gen());
 
         //Set the canvas target and draw on it.
-        tvgexam::verify(canvas->target(instance, surface, WIDTH, HEIGHT));
+        tvgexam::verify(canvas->target(device, instance, renderTarget, SIZE, SIZE, 1));
 
         content(canvas.get());
         if (tvgexam::verify(canvas->draw())) {
             tvgexam::verify(canvas->sync());
         }
-    }
 
+        // After the GlCanvas::sync() function, the content is rendered to the framebuffer
+        // The texture is now ready to be blit to the screen
+        uint32_t y = (counter / NUM_PER_LINE) * SIZE;
+        uint32_t x = (counter % NUM_PER_LINE) * SIZE;
+        wgCopyTextureToTexture(device, renderTarget, surfaceTexture.texture, x, y, SIZE, SIZE);
+    }
+    
     SDL_ShowWindow(window);
+    wgpuSurfacePresent(surface);
 
     mainloop();
 
+    wgpuTextureRelease(surfaceTexture.texture);
+    wgpuTextureDestroy(renderTarget);
+    wgpuTextureRelease(renderTarget);
+    wgpuDeviceRelease(device);
+    wgpuAdapterRelease(adapter);
+    wgpuSurfaceUnconfigure(surface);
+    wgpuSurfaceRelease(surface);
     wgpuInstanceRelease(instance);
     SDL_DestroyWindow(window);
 #else
     cout << "webgpu driver is not detected!" << endl;
```

Comment:
Please retain the original message. This doesn't mean the driver problem, user might build without thorvg webgpu backend. 
```
cout << "Not Support WebGPU" << endl;
```

Reply:
fixed